opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/ChildRegisterFragmentPresenter.java

```diff
@@ -0,0 +1,222 @@
+            Event baseEvent = org.smartregister.util.JsonFormUtils.createEvent(fields, getJSONObject(jsonForm, METADATA), JsonClientProcessingUtils.formTag(allSharedPreferences), entityId, REGISTER_CHILD_EVENT, CHILD_TABLE);
+            JsonClientProcessingUtils.tagSyncMetadata(allSharedPreferences, baseEvent);
+
+            // save the client
+            JSONObject clientJson = new JSONObject(org.smartregister.family.util.JsonFormUtils.gson.toJson(baseClient));
+            getSyncHelper().addClient(baseClient.getBaseEntityId(), clientJson);
+
+            // save the event
+            JSONObject eventJson = new JSONObject(org.smartregister.family.util.JsonFormUtils.gson.toJson(baseEvent));
+            getSyncHelper().addEvent(baseEvent.getBaseEntityId(), eventJson);
+
+            // client process
+
+            org.smartregister.domain.db.Event domainEvent = org.smartregister.family.util.JsonFormUtils.gson.fromJson(eventJson.toString(), org.smartregister.domain.db.Event.class);
+            org.smartregister.domain.db.Client domainClient = org.smartregister.family.util.JsonFormUtils.gson.fromJson(clientJson.toString(), org.smartregister.domain.db.Client.class);
+            EventClient eventClient = new EventClient(domainEvent, domainClient);
+
+            RevealClientProcessor.getInstance(getInstance().getApplicationContext()).processClient(Collections.singletonList(eventClient), true);
+
+            long lastSyncTimeStamp = allSharedPreferences.fetchLastUpdatedAtDate(0);
+            Date lastSyncDate = new Date(lastSyncTimeStamp);
+            allSharedPreferences.saveLastUpdatedAtDate(lastSyncDate.getTime());
+
+
+            // create a task
+            TaskUtils taskUtils = TaskUtils.getInstance();
+            taskUtils.generateDrugAdministrationTask(context, entityId);
+            return true;
+        };
+
+        myInteractor.execute(callable, new CallableInteractorCallBack<Boolean>() {
+            @Override
+            public void onResult(Boolean aBoolean) {
+                ChildRegisterFragmentContract.View view = getView();
+                if (view != null) {
+                    if (aBoolean) {
+                        view.refreshView();
+                    } else {
+                        view.onFetchError(new Exception("An error while saving"));
+                    }
+                    view.setLoadingState(false);
+                }
+            }
+
+            @Override
+            public void onError(Exception ex) {
+                ChildRegisterFragmentContract.View view = getView();
+                if (view == null) return;
+                view.onFetchError(ex);
+                view.setLoadingState(false);
+            }
+        });
+
+    }
+
+    public ECSyncHelper getSyncHelper() {
+        return ECSyncHelper.getInstance(RevealApplication.getInstance().getContext().applicationContext());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'RevealApplication.getInstance' due to existing static import 'org.smartregister.reveal.application.RevealApplication.getInstance'](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5414138)